File: PipelineService/Models/MqttMessages/OperationExecutedMessage.cs
using System;

namespace PipelineService.Models.MqttMessages
{
	public class OperationExecutedMessage : BaseMqttMessage
	{
		/// <summary>
		/// The pipeline's id the node was executed for.
		/// </summary>
		public Guid PipelineId { get; set; }

		/// <summary>
		/// The pipeline execution this node belongs to.
		/// </summary>
		public Guid ExecutionId { get; set; }

		/// <summary>
		/// The operation's id (within a pipeline or execution) that was executed.
		/// </summary>
		public Guid OperationId { get; set; }

		/// <summary>
		/// The operation's id that will be executed.
		/// </summary>
		public Guid WorkerOperationId { get; set; }

		/// <summary>
		/// An identifier of the operation.
		/// Used by generic pandas operation to execute correct function.
		/// </summary>
		public string WorkerOperationIdentifier { get; set; }

		/// <summary>
		/// Indicates if the execution was successful.
		/// </summary>
		public bool Successful { get; set; }

		/// <summary>
		/// A description of the error that occured if the operation could not be executed successfully.
		/// </summary>
		public string ErrorDescription { get; set; }

		/// <summary>
		/// The time (UTC) the execution of this node was started.
		/// </summary>
		public DateTime StartTime { get; set; }

		/// <summary>
		/// The time (UTC) the execution of this node was stopped (either due to an error or completion).
		/// </summary>
		public DateTime StopTime { get; set; }

		/// <summary>
		/// Indicates if the operation has actually been executed or a cached result already exists.
		/// </summary>
		public bool Cached { get; set; } = false;
	}
}

File: OperationWorker/src/models/operation_executed_message.py
import datetime

from src.models.message import Message


class OperationExecutedMessage(Message):
	pipeline_id: str
	execution_id: str
	operation_id: str
	worker_operation_id: str
	worker_operation_identifier: str
	successful: bool
	start_time: datetime
	stop_time: datetime
	error_description: str = None

	def __init__(self):
		super().__init__()

		self.start_time = datetime.datetime.now(datetime.timezone.utc)

	def set_pipeline_id(self, pipeline_id: str):
		self.pipeline_id = pipeline_id

	def set_operation_id(self, operation_id: str):
		self.operation_id = operation_id

	def set_worker_operation_id(self, worker_operation_id: str):
		self.worker_operation_id = worker_operation_id

	def set_worker_operation_identifier(self, worker_operation_id: str):
		self.worker_operation_identifier = worker_operation_id

	def set_execution_id(self, execution_id: str):
		self.execution_id = execution_id

	def set_successful(self, successful: bool):
		self.successful = successful

	def get_successful(self) -> bool:
		return self.successful

	def set_start_time(self, start_time: datetime):
		self.start_time = start_time

	def set_stop_time(self, stop_time: datetime):
		self.stop_time = stop_time

	def set_error_description(self, error_description):
		self.error_description = error_description

	def get_error_description(self) -> str:
		return self.error_description
